src/main/kotlin/no/nav/tilleggsstonader/sak/opplysninger/oppgave/OppgaveService.kt

```diff
@@ -45,13 +46,28 @@ class OppgaveService(
 
     private val logger = LoggerFactory.getLogger(javaClass)
 
-    fun hentOppgaver(finnOppgaveRequest: FinnOppgaveRequest): FinnOppgaveResponseDto {
-        val oppgaveResponse = oppgaveClient.hentOppgaver(finnOppgaveRequest)
+    fun hentOppgaver(finnOppgaveRequest: FinnOppgaveRequestDto): FinnOppgaveResponseDto {
+        FnrUtil.validerOptionalIdent(finnOppgaveRequest.ident)
+
+        val aktørId = finnOppgaveRequest.ident.takeUnless { it.isNullOrBlank() }
+            ?.let { personService.hentAktørIder(it).identer.first().ident }
+
+        val oppgaveResponse =
+            oppgaveClient.hentOppgaver(finnOppgaveRequest.tilFinnOppgaveRequest(aktørId, finnVentemappe()))
+
         val personer = personService.hentPersonKortBolk(oppgaveResponse.oppgaver.mapNotNull { it.ident }.distinct())
         val behandlingIdPåOppgaveId = finnBehandlingId(oppgaveResponse.oppgaver)
 
+        val oppgaver = if (finnOppgaveRequest.oppgaverPåVent) {
+            oppgaveResponse.oppgaver.filter { it.mappeId?.get().let { it == finnVentemappe().id.toLong() } }
+        } else {
+            oppgaveResponse.oppgaver.filterNot { it.mappeId?.get().let { it == finnVentemappe().id.toLong() } }
+        }
+
+        val antallBortfiltrerteOppgaver = oppgaveResponse.oppgaver.size - oppgaver.size
+
         return FinnOppgaveResponseDto(
-            antallTreffTotalt = oppgaveResponse.antallTreffTotalt,
+            antallTreffTotalt = oppgaveResponse.antallTreffTotalt - antallBortfiltrerteOppgaver,
```

Comment:
Dette kan kanskje ødelegge pagineringen litt dersom det er veldig mange oppgaver, og ikke så mange venteoppgaver. Bør vi kanskje sette limit høyere når vi skal hente venteoppgaver?

Reply:
Ja denne delen(endringer i rad 61-70) tror jeg ikke trengs

Tenker man kan bruke erUtenMappe = !this.oppgaverPåVent, i tilFinnOppgaveRequest
Så fjerne valget om å kunne sende inn erUtenMappe fra frontend.

Reply:
Ah, visste ikke at det var sånn det feltet fungerte 🤩  (Det er heller ikke dokumentert i deres docs, så lurer på hvorfor) Men da slipper vi å håndtere filtreringen selv.